gh-69990: Make Profile.print_stats support sorting by multiple values (#104590)
python/cpython: Doc/library/profile.rst

      Create a :class:`~pstats.Stats` object based on the current
      profile and print the results to stdout.

      The *sort* parameter specifies the sorting order of the displayed
      statistics. It accepts a single sort key or a tuple of sort keys
      to enable multi-level sorting.

      .. versionadded:: 3.13

>          :meth:`~Profile.print_stats` can now accept multiple sort arguments.

Reviewer: "can accept multiple sort arguments" looks confusing. Does `print_stats('calls', 'time')` or `print_stats(sort='calls', sort='time')` work?

Author: _Example:_
```python
import cProfile
from pstats import SortKey

with cProfile.Profile() as pr:
    pr.print_stats(sort=("time", "calls"))  # or pr.print_stats(sort=(SortKey.TIME, SortKey.CALLS))
```

I believe that adding a small example to the documentation would make it easier to understand the feature. What do you think?

      Create a :class:`~pstats.Stats` object based on the current
      profile and print the results to stdout.

      The *sort* parameter specifies the sorting order of the displayed

>       statistics. It accepts tuple of SortKey or accepts a tuple of SortKeys
>       to enable multi-level sorting.

Reviewer 2: ```suggestion
      statistics. It accepts a single key or a tuple of keys to enable
      multi-level sorting. A key can be either an integer, a string, or a
      SortKey enum identifying the basis of a sort (for example: -1,
      'time', 'name', SortKey.TIME, or SortKey.NAME).
```

Reviewer 2: This one uses a more apparent structure like the [sort_stats doc](https://docs.python.org/3.13/library/profile.html#pstats.Stats.sort_stats).

Author: Thanks @egeakman !

>       multi-level sorting, as in :func:`Stats.sort_stats <pstats.Stats.sort_stats>.

Reviewer: ```suggestion
      multi-level sorting, as in :func:`Stats.sort_stats <pstats.Stats.sort_stats>`.
```

      .. versionadded:: 3.13

>          :meth:`~Profile.print_stats` now also accepts a tuple of SortKeys.

Reviewer 2: ```suggestion
         :meth:`~Profile.print_stats` now accepts a single key or a tuple of keys.
```

Reviewer 2: I think this revision is easier to understand.

      Create a :class:`~pstats.Stats` object based on the current
      profile and print the results to stdout.

      The *sort* parameter specifies the sorting order of the displayed
      statistics. It accepts a single key or a tuple of keys to enable

>       multi-level sorting. A key can be either an integer, a string, or a

Reviewer: Only a single key may be an integer.

>       SortKey enum identifying the basis of a sort (for example: -1,
>       'time', 'name', SortKey.TIME, or SortKey.NAME).

Reviewer: ```suggestion
      SortKey enum identifying the basis of a sort (for example: ``-1``,
      ``'time'``, ``'name'``, ``SortKey.TIME``, or ``SortKey.NAME``).
```
Or you can just add a reference to `pstats.Stats.sort_stats()`.

>       multi-level sorting. A key can be either an integer, a string, or a
>       SortKey enum identifying the basis of a sort (for example: ``-1``,
>       ``'time'``, ``'name'``, ``SortKey.TIME``, or ``SortKey.NAME``).

Reviewer 2: ```suggestion
      multi-level sorting. When passing a tuple of keys, each key must be
      either a string or a SortKey enum, excluding integers. However,
      if only a single key is provided, it can be an integer, a string, or a
      SortKey enum. These keys serve to identify the basis of the sort,
      such as ``'time'``, ``'name'``, ``SortKey.TIME``, or ``SortKey.NAME``,
      among others."
```

      Create a :class:`~pstats.Stats` object based on the current
      profile and print the results to stdout.

      The *sort* parameter specifies the sorting order of the displayed

>       statistics. It accepts a single key or a tuple of keys to enable
>       multi-level sorting. When passing a tuple of keys, each key must be
>       either a string or a SortKey enum, excluding integers. However,
>       if only a single key is provided, it can be an integer, a string, or a
>       SortKey enum. These keys serve to identify the basis of the sort,
>       such as ``'time'``, ``'name'``, ``SortKey.TIME``, or ``SortKey.NAME``,
>       among others."

Reviewer: ```suggestion
      statistics. It accepts a single key or a tuple of keys to enable
      multi-level sorting, as in :func:`Stats.sort_stats <pstats.Stats.sort_stats>.
```

      .. versionadded:: 3.13

>          :meth:`~Profile.print_stats` now accepts a single key or a tuple of keys.

Reviewer: ```suggestion
         :meth:`~Profile.print_stats` now accepts a tuple of keys.
```
